structopt-derive/src/lib.rs

```diff
@@ -239,6 +239,16 @@ fn gen_augmentation(
 }
 
 fn gen_constructor(fields: &Punctuated<Field, Comma>, parent_attribute: &Attrs) -> TokenStream {
+    // This ident is used in several match branches below,
+    // and the `quote[_spanned]` invocations have different spans.
+    //
+    // Given that this ident is used in several places and
+    // that the branches are located inside of a loop, it is possible that
+    // this ident will be given _different_ spans in different places, and
```

Comment:
Note that a plain `quote { }` or `format_ident` will always use `Span::call_site()` for hygiene, so the generated `matches` token will always be 'the same' from the perspective of the compiler's name resolution. However, this change is still a good idea, since it makes it clear that `matches` is logically independent of the loop.